src/main/java/com/wavefront/sdk/entities/histograms/WavefrontHistogramImpl.java

```diff
@@ -266,19 +178,38 @@ private long currentMinuteMillis() {
   }
 
   /**
-   * Helper to retrieve the current bin. Will be invoked on the thread local histogramBinsList.
+   * Helper to get the current bin.
+   * Will flush currentMinuteBin into globalHistogramBinsList if it's a new minute.
    */
-  private MinuteBin getCurrentBin() {
-    ConcurrentLinkedDeque<MinuteBin> sharedBinsInstance = histogramBinsList.get();
+  private ThreadMinuteBin getCurrentBin() {
     long currMinuteMillis = currentMinuteMillis();
-    if (sharedBinsInstance.isEmpty() ||
-        sharedBinsInstance.getLast().minuteMillis != currMinuteMillis) {
-      sharedBinsInstance.add(new MinuteBin(ACCURACY, currMinuteMillis));
-      if (sharedBinsInstance.size() > MAX_BINS) {
-        sharedBinsInstance.removeFirst();
+    return flushCurrentBin(currMinuteMillis);
+  }
+
+  private ThreadMinuteBin flushCurrentBin(long currMinuteMillis) {
```

Comment:
getOrUpdateCurrentBin is a better method name which conveys the meaning better.
We are not really flushing the bin here so the name does not seem correct to me.